### PR TITLE
feat: frontmatter

### PR DESCRIPTION
**Pull request by P0lip**
TODO:
- [x] ~export `YamlParserResult` type in `@stoplight/yaml` (yaml-as-parser dep is needed due to the lack of this type)~ (not needed, we have js-yaml in place and it's sufficient)
- [x] stringify method
- [x] should update positional info

Nice to have:
- ~indentation detection~ (not needed, stringify takes care of it)
- a few more tests would be nice to have
- own directory

### CODE REVIEW

File: src/frontmatter/frontmatter.ts [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+
+  public get(prop: keyof T): T[keyof T] | void {
+    if (this.properties !== null) {
+      return this.properties[prop];
+    }
+  }
+
+  public set(prop: keyof T, value: T[keyof T]) {
+    if (this.properties !== null) {
+      this.properties[prop] = value;
```
Comment (lott-ai) [inline src/frontmatter/frontmatter.ts]: Can we use lodash [get](https://www.npmjs.com/package/lodash.get), [set](https://www.npmjs.com/package/lodash.set), and [unset](https://www.npmjs.com/package/lodash.unset) here instead?

For example on the homepage/blog, we have some pretty deeply nested frontmatter objects 😬 

https://raw.githubusercontent.com/stoplightio/stoplight.io/develop/netlify/blog-posts/10-000-apis-under-the-sea.md

**Review by lott-ai — CHANGES_REQUESTED** [review]

File: src/frontmatter/frontmatter.ts (hunk 1 above)
Comment (P0lip) [reply src/frontmatter/frontmatter.ts]: @lottamus sure!
I was unaware of the case you mentioned, so did in a more traditional fashion.

Comment (lott-ai) [reply src/frontmatter/frontmatter.ts]: no worries! i think we will be able to use this for the homepage too when we have our own admin panel so might make sense to add it. those 3 lodash functions should be a small dep, right?

Comment (P0lip) [reply src/frontmatter/frontmatter.ts]: Yup, it shouldn't add any significant overhead. I believe Lodash is likely to be already included anyway by one of our dependencies.

File: src/frontmatter/frontmatter.ts [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+    }
+  }
+
+  public unset(prop: PropertyPath) {
+    if (this.properties !== null) {
+      const path = _toPath(prop);
+      const lastSegment = Number(path[path.length - 1]);
+      if (!Number.isNaN(lastSegment)) {
+        const baseObj = path.length > 1 ? this.get(path.slice(0, path.length - 1)) : this.getAll();
+        if (!Array.isArray(baseObj) || baseObj.length < lastSegment) return;
```
Comment (lott-ai) [inline src/frontmatter/frontmatter.ts]: If it's not an array, should we just do the lodash unset?

**Review by lott-ai — APPROVED** [review]

File: src/frontmatter/frontmatter.ts (hunk 2 above)
Comment (P0lip) [reply src/frontmatter/frontmatter.ts]: True, there might be an object literal containing numeric literal as a key.

**Pull request merged**